src/main/scala/TableGroup.scala

```diff
@@ -423,4 +425,13 @@ case class TableGroup private (
       case (_, table) => table.warnings
     })
   warnings = warnings.concat(warningsFromTables)
+
+  def validateRow(row: CSVRecord, tableUrl: String) = {
+    // revisit if we need a variable validatedTables (hash) as in csvlint
+    tables(tableUrl).validateRow(row)
```

Comment:
Whilst the performance of looking objects up in a dictionary shouldn't be *too* bad, it's still better to only look them up once instead of one per row. 

Perhaps we could re-write the validator so that it calls `tableGroup.getTable(tableUrl)` and then directly calls `table.validateRow(row)` itself? That way it can keep a hold of which table it is and avoid having to look it up thousands of times.